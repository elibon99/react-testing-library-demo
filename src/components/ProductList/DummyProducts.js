import Macbook from "../../assets/images/Macbook.jpg"
import Iphone from "../../assets/images/Iphone.jpg"
import AirPods from "../../assets/images/AirPods.webp"

export const DummyProducts = [
    {
        id: 1,
        title: "Macbook",
        price: 16000,
        image: Macbook
    },
    {
        id: 2,
        title: "Iphone",
        price: 7000,
        image: Iphone
    },
    {
        id: 3,
        title: "AirPods",
        price: 1500,
        image: AirPods
    }
]