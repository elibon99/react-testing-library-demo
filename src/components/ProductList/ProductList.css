
.product-list-container {
    display: flex;
    flex-direction: column;
    background: white;
    border-radius: 8px;
    width: 65%;
    padding: 30px 30px;
}

.product-list {
    display: flex;
    justify-content: space-between;
    margin-top: 20px;
    align-items: stretch;
}

.product-list-header {
    margin: 0 0;
}