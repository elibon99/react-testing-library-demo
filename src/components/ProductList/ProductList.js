import React from 'react'
import "./ProductList.css"
import {DummyProducts} from "./DummyProducts";
import ProductItem from "../ProductItem/ProductItem";

const ProductList = ({products, setProducts}) => {
    const addProductToCart = (id) => {
        let product = DummyProducts.find((product) => product.id === id)
        setProducts(prev => [...prev, product])
    }

    return (
        <div className="product-list-container">
            <h3 className="product-list-header">Products</h3>
            <div className="product-list" data-testid={"product-list"}>
                {DummyProducts.map((product, index) => {
                    return(
                        <ProductItem key={index} product={product} products={products} addProductToCart={addProductToCart}/>
                    )
                })}
            </div>
        </div>
    )
}

export default ProductList