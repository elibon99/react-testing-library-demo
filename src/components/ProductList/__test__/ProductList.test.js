import {render, screen} from "@testing-library/react";
import ProductList from "../ProductList";
import {DummyProducts} from "../DummyProducts";

const ShoppingCart = DummyProducts.filter((product) => product.title === "Macbook")

describe('<ProductList/>', () => {

    /**
     * Checks that ProductList is rendered correctly by comparing with snapshot
     */
    it('renders correctly', async () => {
        const {asFragment} = render(
            <ProductList
                products={ShoppingCart}
                setProducts={() => {}}
            />
        )
        expect(asFragment()).toMatchSnapshot()
    })
})