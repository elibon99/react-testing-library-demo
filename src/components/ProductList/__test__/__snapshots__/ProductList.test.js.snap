// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductList/> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-list-container"
  >
    <h3
      class="product-list-header"
    >
      Products
    </h3>
    <div
      class="product-list"
      data-testid="product-list"
    >
      <div
        class="product-item-container"
        id="product-item"
      >
        <img
          alt="product"
          class="product-item-image"
          src="Macbook.jpg"
        />
        <h4
          class="product-item-title"
        >
          Macbook
        </h4>
        <h4
          class="product-item-price"
        >
          16000 SEK
        </h4>
        <button
          class="product-item-btn product-item-btn-disabled"
          title="Macbook"
        >
          Add to cart
        </button>
      </div>
      <div
        class="product-item-container"
        id="product-item"
      >
        <img
          alt="product"
          class="product-item-image"
          src="Iphone.jpg"
        />
        <h4
          class="product-item-title"
        >
          Iphone
        </h4>
        <h4
          class="product-item-price"
        >
          7000 SEK
        </h4>
        <button
          class="product-item-btn undefined"
          title="Iphone"
        >
          Add to cart
        </button>
      </div>
      <div
        class="product-item-container"
        id="product-item"
      >
        <img
          alt="product"
          class="product-item-image"
          src="AirPods.webp"
        />
        <h4
          class="product-item-title"
        >
          AirPods
        </h4>
        <h4
          class="product-item-price"
        >
          1500 SEK
        </h4>
        <button
          class="product-item-btn undefined"
          title="AirPods"
        >
          Add to cart
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
